gui/default/syncthing/core/syncthingController.js

```diff
@@ -2887,6 +2887,34 @@ angular.module('syncthing.core')
             return $scope.version.version + ', ' + os + ' (' + arch + ')';
         };
 
+        $scope.versionBase = function () {
+            if (!$scope.version.version) {
+                return '';
+            }
+            var version = $scope.version.version;
+            var pos = version.indexOf('-');
+            if (pos > 0) {
+                version = version.slice(0, pos);
+            }
+            return version;
+        };
+
+        $scope.docsURL = function (path) {
+            var url = 'https://docs.syncthing.net';
+            if (path) {
+                var hash = path.indexOf('#');
+                if (hash != -1) {
+                    url += '/' + path.slice(0, hash);
+                    url += '?version=' + $scope.versionBase();
+                    url += path.slice(hash);
+                } else {
+                    url += '/' + path;
+                    url += '?version=' + $scope.versionBase();
+                }
+            }
+            return url;
+        };
```

Comment:
I realized something interesting about our docs serving at the moment, in that most of the old URLs here ended with `.html` but one didn't, and the one that didn't still worked because apparently Netlify does magic and adds it internally instead of 404:ing. That makes the above code (which omits the `.html`) work, and means we can lay the foundation for a switch to generating nicer docs urls with `make dirhtml` without breaking everything...

Reply:
It is however irreversible in the sense that we'll then have versions out there referring to both /foo and /foo.html and we'll probably need to handle both variants server side for a long while or risk breaking some subset of users.

Reply:
I agree, since there have been many versions now with mixed variants, we're bound to keep the fall-back behavior on the server.  Should be relatively easy to do though on any hosting?  Just needs to be remembered.  And in the end, having some docs links from ancient versions stop working someday isn't the end of the world.  The internal links *within* the docs are correct anyway.  So that's a go in leaving out `.html` from now on?